### ISSUE TITLE
Connect Federal Agency drop down with the federal agency table

### ISSUE THREAD

**Issue by kristinacyin** [cisagov/manage.get.gov#1793]
### Current Behavior
We recently added the federal agencies table, but it is not connected to anything, we should connect it to in the "Organization name and mailing address" (for org type "Federal") request page and anywhere else the federal agency field is used. 
Note in the current state in the "Organization name and mailing address" (for org type "Federal") request page, the current "Federal agency" dropdown shows duplicate entries which do _not_ exist in the federal agency table (they were removed intentionally). There is also an invalid option in the dropdown: "State, Local, and Tribal Government" which should be just Non-Federal Agency instead.  These duplicate and incorrect values do **not** exist in the Federal Agency table; however, there is a migration script ticket (#1975)  to update any 'deprecated' values to the correct ones to use.  If you pick up this current ticket, the migration ticket is a logical choice to work at the same time. They should be merged around the same time.

There are also 2 values we should keep but not allow end users to select:
- gov Administration
- Non-Federal Agency 

### Acceptance Criteria 
- [ ] Make all occurrences of the federal_agency drop down use the new Agency field that is connected to the federal agency table see #2029 and #1896
- [ ] Hide 2 values when shown in the request form's dropdown (but keep them in the database/Django admin):
  - [ ] gov Administration
  - [ ] Non-Federal Agency
- [x] ~In the request form's Federal Agency dropdown, include an "other" field to allow users to add an agency not in our list (ignore this one)~
- [ ] Make sure federal agency on domain request and domain info are all still drop downs
- [ ] ~remove the old federal_agency field  as it is no longer needed and be sure the now linked Agency field now shows on the django admin exactly where federal_agency was showing and that the field is viewable for all users (note the goal is to make it look like nothing changed for analysts)~
- [ ] Create 2 migration files:  1: Migration file to DELETE `federal_agency` 2: Migration file to RENAME `updated_federal_agency` to `federal_agency` (updated_federal_agency now filled in at this point, so we have to delete + rename)
- [ ] Remove tests testing that updated_federal_agency duplicates the FederalAgency version of federal_agency that reference updated_federal_agency as this field will be renamed to federal_agency to replace the current federal_agency field.
- [ ] Make sure both Superuser and Analysts can view the correct new Federal Agency list without issues

### Additional Context

[Slack thread discussion](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1708109328529279)

### Issue Links
blocked by #1896 
blocked by #2029 
Relates to #1975 (do at around the same time to avoid issues)
Relates to: #337

**Comment by h-m-f-t** [comment]
I've refined the text and struck a few things from this task. Added the refinement label to talk about "hiding" two values. 

Before we add an "other" field that prompts for a user-supplied value, let's determine how this selection would flow into the rest of the request form/DjAdmin/notification emails that are sent.  @kristinacyin

**Comment by abroddrick** [comment]
updated language on AC 3

**Comment by h-m-f-t** [comment]
In refinement, we decided to take this opportunity to create a table for federal agencies and to make the domain mapping A/C its own issue. @abroddrick will take a first pass.

**Comment by vickyszuchin** [comment]
Moved status backed to "Planned" to "Ready" since it was moved out from sprint 43 into sprint 44.

**Comment by therealslimhsiehdy** [comment]
This is Ticket #3 for Federal Agency

**Comment by vickyszuchin** [comment]
Removed "blocked" label since the tickets that are blocking this ticket are closed.
